Add new Windstream Parser

aliex-13 opened the pull request:
Raising this Pull Request to onboard a new vendor - Windstream.

All testing I have completed looks to deal with these emails correctly.

Invoke tests --local does report that I have too many local variables but I am not sure how strict we are being on these things as I am seeing messages for other parsers in there as well.

chadell commented on circuit_maintenance_parser/parsers/windstream.py:
```diff
@@ -0,0 +1,77 @@
+            value_tag = cols[1].string.strip()
+            if header_tag == "WMT:":
+                data["maintenance_id"] = value_tag
+            elif header_tag == "Event Start Date & Time:":
+                value_tag = value_tag.replace(" ET", "")
+                value_tag = datetime.strptime(value_tag, "%m/%d/%y %H:%M")
+                data["start"] = int(value_tag.replace(tzinfo=timezone.utc).timestamp())
+            elif header_tag == "Event End Date & Time:":
+                value_tag = value_tag.replace(" ET", "")
+                value_tag = datetime.strptime(value_tag, "%m/%d/%y %H:%M")
+                data["end"] = int(value_tag.replace(tzinfo=timezone.utc).timestamp())
```
what about?
```suggestion
            elif "Date & Time:" in header_tag:
                value_tag = value_tag.replace(" ET", "")
                value_tag = datetime.strptime(value_tag, "%m/%d/%y %H:%M")
                event_time =  int(value_tag.replace(tzinfo=timezone.utc).timestamp())
                if "Event Start" in header_tag:
                    data["start"] = event_time
                elif "Event End" in header_tag:
                    data["end"] = event_time
```

aliex-13 replied:
Yeah that should work! I'll add it now and test it out.

Note: Tested and works fine :)

chadell reviewed (dismissed):
LGTM

chadell commented:
> Invoke tests --local does report that I have too many local variables but I am not sure how strict we are being on these things as I am seeing messages for other parsers in there as well.

Don't worry about it

aliex-13 commented:
Hey @chadell - I just noticed that my start/end logic was incorrect as it was marking the initial time received from the vendor as UTC from the get-go. I have modified the code to correctly localise it to "EST". Conversions to UTC will then occur correctly. This did mean that I needed to move some things around as I added a bunch of extra local variables to deal with it. I have managed to avoid going over the number of local variables (15) and statements in my script (50) due to the things I shuffled in the last commit. Everything tested correctly now and this should be the last commit as long as you are happy with the changes I made!
Thanks :)

chadell commented on circuit_maintenance_parser/parsers/windstream.py:
```diff
@@ -0,0 +1,78 @@
+            elif "Date & Time:" in header_tag:
+                value_tag = value_tag.replace(" ET", "")
+                date_obj_est = datetime.strptime(value_tag, "%m/%d/%y %H:%M")
+                date_obj_est = pytz.timezone("US/Eastern").localize(date_obj_est)
```
I guess that this may change depending on the customer's timezone. what do you think?

aliex-13 replied:
Hmm I guess that is a good point.
I may only have maintenance email examples from Windstream for our circuits that belong in that region. I haven't got any other emails to base this information on.

Do you have any functions which use a timezone checker that I can replicate (based on the abbreviation used at the end of a datetime string)?
If not, I'll come up with my own :)

chadell replied:
no @aliex-13 , we don't have, but I think it would be a useful helper function

aliex-13 replied:
Hey @chadell - I've pushed a new commit with a new helper function in the "utils.py" file. I'm not sure if this was the correct location but I could see similar functions there. Please let me know if I need to move it/fix it.

I've added all of the basic US related timezones to this function. 
We can easily add more as shown in the code.

Does this work for you? All of my tests pass.

chadell commented on circuit_maintenance_parser/parsers/windstream.py:
```diff
@@ -0,0 +1,78 @@
+        table = soup.find("table", "circuitTable")
+        for row in table.find_all("tr"):
+            cols = row.find_all("td")
+            if len(cols) == 9:
```
Do you think that the table will have always 9 columns? could we make this a bit more "robust"?

aliex-13 replied:
We already make it a bit more robust by checking that it is a 'circuitTable'. From all the emails I have with Windstream, they keep those headers consistent and there is always 9 columns.

Do you have any recommendations to make this more robust rather than checking the length of the fields that we expect to be there?

chadell reviewed (approved):
LGTM

Thanks @aliex-13 for your awesome contribution!
